xilem_core/src/state.rs

```diff
@@ -203,17 +206,32 @@ impl<T0: ViewArgument, T1: ViewArgument> ViewArgument for (T0, T1) {
     }
 }
 
-// TODO: Generate 2+ with a macro; maybe 1+, but then again for understandability purposes, having at least one
-// outside of the macro is appealing..
-impl<T0: ViewArgument, T1: ViewArgument, T2: ViewArgument> ViewArgument for (T0, T1, T2) {
-    type Params<'a> = (T0::Params<'a>, T1::Params<'a>, T2::Params<'a>);
-    fn reborrow_mut<'input, 'a: 'input>(
-        (t0, t1, t2): &'input mut Self::Params<'a>,
-    ) -> Self::Params<'input> {
-        (
-            T0::reborrow_mut(t0),
-            T1::reborrow_mut(t1),
-            T2::reborrow_mut(t2),
-        )
-    }
+// We use manual impls for 0, 1 and 2 to show the pattern
+viewargument_tuple!(T0, T1, T2);
+viewargument_tuple!(T0, T1, T2, T3);
+viewargument_tuple!(T0, T1, T2, T3, T4);
+viewargument_tuple!(T0, T1, T2, T3, T4, T5);
+viewargument_tuple!(T0, T1, T2, T3, T4, T5, T6);
+viewargument_tuple!(T0, T1, T2, T3, T4, T5, T6, T7);
```

Comment:
Is there a reason why we "only" support 8 arguments? (I think 12 would be a reasonable value to be in sync with std trait impls like `Default` or `Debug` ?)